pytorch_lightning/plugins/training_type/deepspeed.py

```diff
@@ -430,8 +431,9 @@ def _initialize_deepspeed_train(self, model):
 
         # although we set these here, deepspeed manages the specific optimizer logic
         self.lightning_module.trainer.optimizers = [deepspeed_optimizer]
-        lr_scheduler["scheduler"] = deepspeed_scheduler
-        self.lightning_module.trainer.lr_schedulers = [lr_scheduler]
+        if scheduler:
```

Comment:
should this be?:

```suggestion
        if deepspeed_scheduler is not None:
```

we check that `"optimizer" in self.config` but not the scheduler key.

Reply:
I think you are correct @carmocca as this ensures that if the user defines a scheduler within the config object, this is referenced lr_scheduler

Reply:
Another test should be added that ensures if a user passes a scheduler via the deepspeed config, this is correctly set